src/pages/ExternalSupplierInvoicePage.js

```diff
@@ -0,0 +1,444 @@
+  ITEM_SELECT: 'itemSelect',
+};
+
+export class ExternalSupplierInvoicePage extends GenericPage {
+  constructor(props) {
+    super(props);
+    this.state.sortBy = 'itemName';
+    this.state.totalPrice = 0;
+    this.state.expiryModelIsOpen = false;
+    this.state.columns = [
+      {
+        key: 'itemCode',
+        width: 2,
+        title: tableStrings.item_code,
+        sortable: true,
+      },
+      {
+        key: 'itemName',
+        width: 4,
+        title: tableStrings.item_name,
+        sortable: true,
+      },
+      {
+        key: 'packSize',
+        width: 1.3,
+        title: tableStrings.pack_size,
+        alignText: 'center',
+      },
+      {
+        key: 'numberOfPacks',
+        width: 1.3,
+        title: tableStrings.quantity,
+        alignText: 'center',
+      },
+      {
+        key: 'batch',
+        width: 1.8,
+        title: tableStrings.batch_name,
+        alignText: 'center',
+      },
+      {
+        key: 'expiryDate',
+        width: 2,
+        title: tableStrings.batch_expiry,
+        alignText: 'center',
+      },
+      {
+        key: 'costPrice',
+        width: 1.4,
+        title: tableStrings.batch_cost_price,
+        alignText: 'center',
+      },
+      {
+        key: 'remove',
+        width: 1,
+        title: tableStrings.remove,
+        alignText: 'center',
+      },
+    ];
+    this.dataTypesSynchronised = DATA_TYPES_SYNCHRONISED;
+    this.finalisableDataType = 'Transaction';
+    this.getFilteredSortedData = this.getFilteredSortedData.bind(this);
+    this.onEndEditing = this.onEndEditing.bind(this);
+    this.onDatabaseEvent = this.onDatabaseEvent.bind(this);
+    this.openItemSelector = this.openItemSelector.bind(this);
+    this.openCommentEditor = this.openCommentEditor.bind(this);
+    this.openTheirRefEditor = this.openTheirRefEditor.bind(this);
+    this.getModalTitle = this.getModalTitle.bind(this);
+    this.addNewLine = this.addNewLine.bind(this);
+  }
+  /**
+   * Returns updated data according to searchTerm, sortBy and isAscending.
+   * Dealing with TransactionBatch as oppose to TransactionItems
+   * to be able to add miltiple batches for item
+   * also total price is calculated here
+   */
+  getFilteredSortedData(searchTerm, sortBy, isAscending) {
+    const { database, transaction } = this.props;
+    console.log(transaction.items);
+    const transactionBatches = transaction.transactionBatches(database)
+      .filtered('itemName BEGINSWITH[c] $0', searchTerm);
+
+    // check to see if transactionBatches exist
+    if (transactionBatches.length === 0) {
+      this.setState({ totalPrice: 0 });
+      return [];
+    }
+
+    let sortDataType;
+    switch (sortBy) {
+      case 'itemName':
+      case 'itemCode':
+      case 'batch':
+        sortDataType = 'string';
+        break;
+      case 'numberOfPacks':
+      case 'packSize':
+        sortDataType = 'number';
+        break;
+      default:
+        sortDataType = 'realm';
+    }
```

Comment:
My preference in situations where you need to get a single value based on a key is to use a dictionary (js object) rather than a switch case.
```
const sortDataType = {
  itemName: 'string',
  itemCode: 'string',
  ...
}[sortBy] || 'realm'; // Use realm as the default sortDataType
```